Bottom: 259150d5ba4a31c22d5f4e23c73dd3f9211ed7ae
Top:    97df7775911b302f295c23a25559d025fd1923b2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 17:29:00 +0000

Refresh of hid-gcode-fix-memory-leak-in-e

---

diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 40a3b36..17bbae1 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -558,6 +558,8 @@ gcode_do_export (HID_Attr_Val * options)
           if (!gcode_f2)
             {
               perror (filename);
+              free (filename);
+              bm_free (bm);
               return;
             }
           fprintf (gcode_f2, "(Created by G-code exporter)\n");
